TransIT.API/Controllers/ActionTypeController.cs

```diff
@@ -1,47 +1,99 @@
 ﻿using System.Threading.Tasks;
-using AutoMapper;
 using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
 using Microsoft.AspNetCore.Mvc;
-using TransIT.BLL.Services;
-using TransIT.BLL.Services.Interfaces;
 using TransIT.BLL.DTOs;
-using TransIT.DAL.Models.Entities;
+using TransIT.BLL.Factory;
 
 namespace TransIT.API.Controllers
 {
+    [ApiController]
+    [EnableCors("CorsPolicy")]
+    [Produces("application/json")]
+    [Route("api/v1/[controller]")]
     [Authorize(Roles = "ADMIN,ENGINEER,REGISTER,ANALYST")]
-    public class ActionTypeController : DataController<ActionType, ActionTypeDTO>
+    public class ActionTypeController : FilterController<ActionTypeDTO>
     {
-        private readonly IActionTypeService _actionTypeService;
+        private readonly IServiceFactory _serviceFactory;
 
-        public ActionTypeController(
-            IMapper mapper,
-            IActionTypeService actionType,
-            IFilterService<ActionType> odService
-            ) : base(mapper, actionType, odService)
+        public ActionTypeController(IServiceFactory serviceFactory, IFilterServiceFactory filterServiceFactory)
+            : base(filterServiceFactory)
         {
-            _actionTypeService = actionType;
+            _serviceFactory = serviceFactory;
         }
```

Comment:
- in the controllers please assign only services that needed from factory
eg
` private readonly IActionTypeService _actionTypeService;


         public AuthenticationController(IAuthenticationService authenticationService) =>	        public AuthenticationController(IServiceFactory serviceFactory)
            _authenticationService = authenticationService;	        {

             _actionTypeService = serviceFactory.ActionTypeService;
        }

do this for all controllers
`